### PR TITLE
passing superagent as config option.

### PR DESCRIPTION
**Pull request by vijayst**

### CODE REVIEW

File: src/api.js [hunk 1]
```diff
@@ -23,14 +23,16 @@ const API = {
...
+      'superagent'
     ]))
   },
-	
+
 	request(options = {}) {
 
 		let { method, data, endpoint, onSuccess, onError, fileData } = options;
     let requestData, requestHeaders, doRequest;
+    const request = this.superagent || superagent;
```
Comment (ikido) [inline src/api.js]: you have to pass superagent instance in options first and include it in destructing assignment, then assign `this.superagent` variable

Comment (vijayst) [inline src/api.js]: I am checking for  superagent object in the request options first, then in config and if none found, use the superagent import.

Comment (ikido) [inline src/api.js]: Yes Vijay, but you are breaking existing code style — variables that can be passed as options are explicitly assigned in destructive assignment above, it is a way of documenting code, to be aware of possible options. superagent option is not assigned that way, that's what I'm complaining about. Thank you for PR btw =)

**Comment by vijayst** [comment]
ok. let me do it using destructive assignment.

**Comment by ikido** [comment]
Thank you for your work Vijay! and for idea about superagent-mock

**Comment by ikido** [comment]
@vijayst fyi I published new version

**Pull request merged**